subsys/bluetooth/host/conn.c

```diff
@@ -2167,20 +2216,8 @@ struct bt_conn *bt_conn_create_le(const bt_addr_le_t *peer,
 
 	conn = bt_conn_lookup_addr_le(BT_ID_DEFAULT, peer);
 	if (conn) {
-		switch (conn->state) {
-		case BT_CONN_CONNECT_SCAN:
-			bt_conn_set_param_le(conn, param);
-			return conn;
-		case BT_CONN_CONNECT:
-		case BT_CONN_CONNECTED:
-			return conn;
-		case BT_CONN_DISCONNECTED:
-			BT_WARN("Found valid but disconnected conn object");
-			goto start_scan;
-		default:
-			bt_conn_unref(conn);
-			return NULL;
-		}
+		bt_conn_unref(conn);
```

Comment:
Maybe add a BT_WARN() here "existing connection object already found" or something like that.

Reply:
Added warning printing the state string. Plus commented about the different states.